cmd/ssh_key_register.go

```diff
@@ -42,37 +41,40 @@ func (c *computeSSHKeyRegisterCmd) cmdPreRun(cmd *cobra.Command, args []string)
 }
 
 func (c *computeSSHKeyRegisterCmd) cmdRun(cmd *cobra.Command, _ []string) error {
-	var (
-		sshKey *egoscale.SSHKey
-		err    error
-	)
-
 	// Template registration can take a _long time_, raising
 	// the Exoscale API client timeout as a precaution.
-	globalstate.EgoscaleClient.SetTimeout(30 * time.Minute)
+	client := globalstate.EgoscaleV3Client.WithHttpClient(&http.Client{Timeout: 30 * time.Minute})
 
-	ctx := exoapi.WithEndpoint(
-		gContext,
-		exoapi.NewReqEndpoint(account.CurrentAccount.Environment, account.CurrentAccount.DefaultZone),
-	)
+	ctx := gContext
 
 	publicKey, err := os.ReadFile(c.PublicKeyFile)
 	if err != nil {
 		return err
 	}
 
-	decorateAsyncOperation(fmt.Sprintf("Registering SSH key %q...", c.Name), func() {
-		sshKey, err = globalstate.EgoscaleClient.RegisterSSHKey(ctx, account.CurrentAccount.DefaultZone, c.Name, string(publicKey))
+	registerKeyRequest := v3.RegisterSSHKeyRequest{
+		Name:      c.Name,
+		PublicKey: string(publicKey),
+	}
+
+	err = decorateAsyncOperations(fmt.Sprintf("Registering SSH key %q...", c.Name), func() error {
+		op, err := client.RegisterSSHKey(ctx, registerKeyRequest)
+		if err != nil {
+			return err
```

Comment:
in the function, we usually add more error wrapping the same way you did in this [file](https://github.com/mlec1/cli/blob/00d437ddb0ae16e2b90fcccf1c166c5ee88d6308/cmd/anti_affinity_group_create.go#L57)